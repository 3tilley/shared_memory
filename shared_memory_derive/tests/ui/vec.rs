//! Vecs and slices are NOT allowed in shared memory

use shared_memory::SharedMemCast;

#[derive(SharedMemCast)]
pub struct MyStruct {
    x: Vec<u32>,
}

#[derive(SharedMemCast)]
pub struct MyStruct2(Vec<u32>);

#[derive(SharedMemCast)]
pub struct MyStructArray<'a> {
    x: &'a [u32],
}

#[derive(SharedMemCast)]
pub struct MyStruct2Array<'a>(&'a [u32]);

fn main() {}
