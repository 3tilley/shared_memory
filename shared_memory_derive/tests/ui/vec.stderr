error[E0277]: the trait bound `std::vec::Vec<u32>: shared_memory::cast::SharedMemCast` is not satisfied
 --> $DIR/vec.rs:7:8
  |
7 |     x: Vec<u32>,
  |        ^^^ the trait `shared_memory::cast::SharedMemCast` is not implemented for `std::vec::Vec<u32>`
  |
  = note: required by `shared_memory::cast::AssertIsSharedMemCast`

error[E0277]: the trait bound `std::vec::Vec<u32>: shared_memory::cast::SharedMemCast` is not satisfied
  --> $DIR/vec.rs:11:22
   |
11 | pub struct MyStruct2(Vec<u32>);
   |                      ^^^ the trait `shared_memory::cast::SharedMemCast` is not implemented for `std::vec::Vec<u32>`
   |
   = note: required by `shared_memory::cast::AssertIsSharedMemCast`

error[E0277]: the trait bound `&'a [u32]: shared_memory::cast::SharedMemCast` is not satisfied
  --> $DIR/vec.rs:15:8
   |
15 |     x: &'a [u32],
   |        ^
   |        |
   |        the trait `shared_memory::cast::SharedMemCast` is not implemented for `&'a [u32]`
   |        help: consider removing 1 leading `&`-references
   |
   = help: the following implementations were found:
             <[T; 1024] as shared_memory::cast::SharedMemCast>
             <[T; 1048576] as shared_memory::cast::SharedMemCast>
             <[T; 1073741824] as shared_memory::cast::SharedMemCast>
             <[T; 10] as shared_memory::cast::SharedMemCast>
           and 55 others
   = note: required by `shared_memory::cast::AssertIsSharedMemCast`

error[E0277]: the trait bound `&'a [u32]: shared_memory::cast::SharedMemCast` is not satisfied
  --> $DIR/vec.rs:19:31
   |
19 | pub struct MyStruct2Array<'a>(&'a [u32]);
   |                               ^
   |                               |
   |                               the trait `shared_memory::cast::SharedMemCast` is not implemented for `&'a [u32]`
   |                               help: consider removing 1 leading `&`-references
   |
   = help: the following implementations were found:
             <[T; 1024] as shared_memory::cast::SharedMemCast>
             <[T; 1048576] as shared_memory::cast::SharedMemCast>
             <[T; 1073741824] as shared_memory::cast::SharedMemCast>
             <[T; 10] as shared_memory::cast::SharedMemCast>
           and 55 others
   = note: required by `shared_memory::cast::AssertIsSharedMemCast`

For more information about this error, try `rustc --explain E0277`.
